project-chip/connectedhomeip
# [Draft] Re-Implement App Install flow

Re-Implementing application installation flow. Continuation of https://github.com/project-chip/connectedhomeip/pull/33959


#### Problem
Current application installation flow blocks commissioning and has too crude functionality that cannot provide great UX (providing installation percentage etc.)

#### Change overview
- Updated pre-commissioning part of app install flow and removed blocking of commissioning if app is not installed
- Add app when it is not installed
- Extend application basic cluster with: installed, not installed, installation failed and installation completed

#### Tests
1. Build the TV app using command:
`scripts/examples/gn_build_example.sh examples/tv-app/linux out/debug/`

2. Build the TV casting app using command:
`scripts/examples/gn_build_example.sh examples/tv-casting-app/linux out/debug/`

3. Start TV App
`out/debug/chip-tv-app`

4. Start TV Casting App
`out/debug/chip-tv-casting-app`

5. Send Application Launcher command
`cast cluster applicationlauncher launch-app 0 1 --Application {"catalogVendorID":123,"applicationID":"com.example.contentapp"}`

You should receive pending state:
`{
    status: 3,
    data: ""
}`

6. Start the app install of content app and send application launcher command again
You should receive installing state:
`{
    status: 4,
    data: ""
}`

7. In case app install failed and you send application launcher command, you should then receive the state:
`{
    status: 3,
    data: "App install failed. Try again"
}`

---------- **TBA** --------
Need to add support for tv linux
